lib/chef/provider/package/yum.rb

```diff
@@ -1081,22 +1082,24 @@ def load_current_resource
                 @new_resource.version($2)
               end
             end
-          end
-
-          if @new_resource.version
-            new_resource = "#{@new_resource.package_name}-#{@new_resource.version}#{yum_arch}"
+            @candidate_version << @new_resource.version
+            installed_version << @yum.installed_version(@current_resource.package_name, arch)
```

Comment:
Generally, initializers should be lightweight, not do much and not raise exceptions.  That gets a little more important because of things like minitest-chef-handler that like to create chef resources and providers to do its work, and its caused issues in the past.